### PR TITLE
Fixes documentation build docs #11972 issue

### PR DESCRIPTION
**Pull request by HimanshuGarg47**
**Issue** #11972 
Summary: Building the docs, while resulting in a working build, raises some errors due to missing Python reference targets.

**Solution**:
This PR has resolved some docs errors that are because of incorrect reference targets using the Sphinx reference directive (e.g. {class}).

**Assumption**:
Unresolved errors are because of fast forward reference and my unknown reason. 

```
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: django.core.files.base.File
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: django.core.files.base.File
wagtail/docs/reference/contrib/routablepage.md:: WARNING: py:attr reference target not found: wagtail.models.Page.url
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: HttpRequest
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: RouteResult
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.find_for_request:1: WARNING: py:class reference target not found: HttpRequest
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.update_aliases:1: WARNING: py:class reference target not found: PageRevision
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.update_aliases:1: WARNING: py:class reference target not found: optional
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.FieldBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.ChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.MultipleChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.PageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/documents/blocks.py:docstring of wagtail.documents.blocks.DocumentChooserBlock:1: WARNING: py:class reference target not found: wagtail.documents.views.chooser.BaseDocumentChooserBlock
wagtail/wagtail/images/blocks.py:docstring of wagtail.images.blocks.ImageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/snippets/blocks.py:docstring of wagtail.snippets.blocks.SnippetChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/blocks/static_block.py:docstring of wagtail.blocks.static_block.StaticBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/struct_block.py:docstring of wagtail.blocks.struct_block.StructBlock:1: WARNING: py:class reference target not found: wagtail.blocks.struct_block.BaseStructBlock
wagtail/wagtail/blocks/list_block.py:docstring of wagtail.blocks.list_block.ListBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/stream_block.py:docstring of wagtail.blocks.stream_block.StreamBlock:1: WARNING: py:class reference target not found: 
```

### CODE REVIEW

File: docs/releases/1.11.rst [hunk 1]
```diff
@@ -121,8 +121,8 @@ to:
 
 and passing those through at the point where you are calling ``get_url_parts`` on ``super`` (if applicable).
 
-See also: :meth:`wagtail.core.models.Page.get_url_parts`, :meth:`wagtail.core.models.Page.get_url`,
-:meth:`wagtail.core.models.Page.get_full_url`, and :meth:`wagtail.core.models.Page.relative_url`
+See also: `wagtail.core.models.Page.get_url_parts`, `wagtail.core.models.Page.get_url`,
+`wagtail.core.models.Page.get_full_url`, and `wagtail.core.models.Page.relative_url`
```
Comment (laymonage) [inline docs/releases/1.11.rst:125]: `wagtail.core.models` has been moved to `wagtail.models`, and these methods are still documented

```suggestion
See also: :meth:`wagtail.models.Page.get_url_parts`, :meth:`wagtail.models.Page.get_url`,
:meth:`wagtail.models.Page.get_full_url`, and :meth:`wagtail.models.Page.relative_url`
```

File: docs/releases/1.2.rst [hunk 2]
```diff
@@ -115,7 +115,7 @@ Bug fixes
 * Dropdown for ``ForeignKey`` s are now styled consistently (Ashia Zawaduk)
 * Date choosers now appear on top of StreamField menus (Sergey Nikitin)
 * Fixed a migration error that was raised when block-updating from 0.8 to 1.1+
-* :meth:`Page.copy() <wagtail.core.models.Page.copy>` no longer breaks on models with a ``ClusterTaggableManager`` or ``ManyToManyField``
+* `Page.copy() <wagtail.core.models.Page.copy>` no longer breaks on models with a ``ClusterTaggableManager`` or ``ManyToManyField``
```
Comment (laymonage) [inline docs/releases/1.2.rst:118]:
```suggestion
* :meth:`Page.copy() <wagtail.models.Page.copy>` no longer breaks on models with a ``ClusterTaggableManager`` or ``ManyToManyField``
```

File: docs/releases/1.6.1.rst [hunk 3]
```diff
@@ -22,7 +22,7 @@ Bug fixes
 ~~~~~~~~~
 
 * Wagtail's middleware classes are now compatible with Django 1.10's `new-style middleware <https://docs.djangoproject.com/en/stable/releases/1.10/#new-style-middleware>`_ (Karl Hobley)
-* The :meth:`~wagtail.core.models.Page.can_create_at` method is now checked in the create page view (Mikalai Radchuk)
+* The `~wagtail.core.models.Page.can_create_at` method is now checked in the create page view (Mikalai Radchuk)
```
Comment (laymonage) [inline docs/releases/1.6.1.rst:25]:
```suggestion
* The :meth:`~wagtail.models.Page.can_create_at` method is now checked in the create page view (Mikalai Radchuk)
```

File: docs/releases/1.8.rst [hunk 4]
```diff
@@ -70,7 +70,7 @@ Other features
 * ``ChoiceBlock`` now accepts a callable as the choices list (Mikalai Radchuk)
 * Redundant action buttons are now omitted from the root page in the explorer (Nick Smith)
 * Locked pages are now disabled from editing at the browser level (Edd Baldry)
-* Added :meth:`wagtail.core.query.PageQuerySet.in_site` method for filtering page QuerySets to pages within the specified site (Chris Rogers)
+* Added `wagtail.core.query.PageQuerySet.in_site` method for filtering page QuerySets to pages within the specified site (Chris Rogers)
```
Comment (laymonage) [inline docs/releases/1.8.rst:73]: https://docs.wagtail.org/en/stable/reference/pages/queryset_reference.html#wagtail.query.PageQuerySet.in_site

```suggestion
* Added :meth:`wagtail.query.PageQuerySet.in_site` method for filtering page QuerySets to pages within the specified site (Chris Rogers)
```

File: docs/releases/5.2.md [hunk 5]
```diff
@@ -63,7 +63,7 @@ Thank you to core contributor LB (Ben) Johnston for writing this documentation.
 
 ### `ModelViewSet` improvements
 
-Several features from {class}`~wagtail.snippets.views.snippets.SnippetViewSet` have been implemented in {class}`~wagtail.admin.viewsets.model.ModelViewSet`, allowing you to use them without registering your models as snippets. For more details on using `ModelViewSet`, refer to [](generic_views).
+Several features from `~wagtail.snippets.views.snippets.SnippetViewSet` have been implemented in `~wagtail.admin.viewsets.model.ModelViewSet`, allowing you to use them without registering your models as snippets. For more details on using `ModelViewSet`, refer to [](generic_views).
```
Comment (laymonage) [inline docs/releases/5.2.md:66]: Are you sure these caused an issue? These are documented and are still the correct paths

File: docs/releases/5.2.md [hunk 6]
```diff
@@ -300,7 +300,7 @@ def register_frank_menu_item():
 
 ### Edit and delete URLs in `ModelViewSet` changed to allow non-integer primary keys
 
-To accommodate models with non-integer primary keys, the URL patterns for the edit and delete views in {class}`~wagtail.admin.viewsets.model.ModelViewSet` have been changed.
+To accommodate models with non-integer primary keys, the URL patterns for the edit and delete views in `~wagtail.admin.viewsets.model.ModelViewSet` have been changed.
```
Comment (laymonage) [inline docs/releases/5.2.md:303]: This too, it's a valid link

File: docs/releases/5.2.md [hunk 7]
```diff
@@ -310,7 +310,7 @@ If you use {func}`~django.urls.reverse` with {meth}`~wagtail.admin.viewsets.base
 
 Redirects for the legacy URLs are in place for backwards compatibility, but will be removed in a future release.
 
-The URLs for snippets underwent similar changes in Wagtail 2.14. The redirects for the legacy URLs in {class}`~wagtail.snippets.views.snippets.SnippetViewSet` have now been marked for removal in a future release.
+The URLs for snippets underwent similar changes in Wagtail 2.14. The redirects for the legacy URLs in `~wagtail.snippets.views.snippets.SnippetViewSet` have now been marked for removal in a future release.
```
Comment (laymonage) [inline docs/releases/5.2.md:313]: Here too

File: wagtail/documents/__init__.py [hunk 8]
```diff
@@ -14,7 +14,7 @@ def get_document_model_string():
 def get_document_model():
     """
     Get the document model from the ``WAGTAILDOCS_DOCUMENT_MODEL`` setting.
-    Defaults to the standard :class:`~wagtail.documents.models.Document` model
+    Defaults to the standard `~wagtail.documents.models.Document` model
```
Comment (laymonage) [inline wagtail/documents/__init__.py:17]: Docstrings still use RST and not markdown, so we need double backticks. We should also remove the `~` because it's part of the `:class:` syntax to make it a shorthand, which is no longer used if we remove the `:class:`.

```suggestion
    Defaults to the standard ``wagtail.documents.models.Document`` model
```

File: wagtail/images/__init__.py [hunk 9]
```diff
@@ -15,7 +15,7 @@ def get_image_model():
     """
     Get the image model from the ``WAGTAILIMAGES_IMAGE_MODEL`` setting.
     Useful for developers making Wagtail plugins that need the image model.
-    Defaults to the standard :class:`~wagtail.images.models.Image` model
+    Defaults to the standard `~wagtail.images.models.Image` model
```
Comment (laymonage) [inline wagtail/images/__init__.py:18]:
```suggestion
    Defaults to the standard ``wagtail.images.models.Image`` model
```

File: docs/conf.py [hunk 10]
```diff
@@ -35,6 +35,7 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 sys.path.insert(0, os.path.abspath(".."))
+sys.path.insert(0, os.path.abspath("../i18n.py"))
```
Comment (laymonage) [inline docs/conf.py:38]: Why was this added? 🤔

**Review by laymonage — COMMENTED** [review]
Thanks for giving this a go!

Some of the references removed in this PR are still valid, and some are just moved to a different paths but they still exist, so they should be fixed to the new paths.

Also, make sure that we use double backticks when working with RST (in docstrings, `.rst` files, or an `eval-rst` block in `.md` files).

Note that the `~` in references are used to shorten the rendered reference to only use the final part of the path, e.g. <code>{class}\`~wagtail.admin.viewsets.model.ModelViewSet\`</code> will be rendered as [`ModelViewSet`](https://docs.wagtail.org/en/stable/reference/viewsets.html#wagtail.admin.viewsets.model.ModelViewSet). When removing a reference, you should either remove the `~` or remove the complete path and just use the final part – you might need to read the documentation's context to decide which one is best.

There are also unrelated changes which should be removed.

As for the remaining errors,

```
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.AbstractRendition
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
```

Some of the above are documented e.g. https://docs.wagtail.org/en/stable/advanced_topics/images/renditions.html#wagtail.images.models.AbstractImage.create_renditions, but unfortunately the docs are scattered across different files

https://github.com/wagtail/wagtail/blob/a7798c1b806fb1c7cfdf5c9fcb80dae3133a759d/docs/advanced_topics/images/renditions.md?plain=1#L156-L175

https://github.com/wagtail/wagtail/blob/a7798c1b806fb1c7cfdf5c9fcb80dae3133a759d/docs/advanced_topics/images/custom_image_model.md?plain=1#L77-L90

I don't think we can remove `:noindex:` in both places without triggering a duplicate warning. Looking at https://github.com/sphinx-doc/sphinx/issues/7052 maybe we should use `:noindexentry:` instead?

---

```
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: django.core.files.base.File
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: django.core.files.base.File
```

The `File` class is documented under `django.core.files.File`: https://docs.djangoproject.com/en/5.0/ref/files/file/#django.core.files.File, so we need to update the docstring in `wagtail.images.models.AbstractImage.generate_rendition_file` to reflect that.

```
wagtail/docs/reference/contrib/routablepage.md:: WARNING: py:attr reference target not found: wagtail.models.Page.url
```

The documentation for `Page.url` was removed in 8d786c35349f097b1c9c8cdc5c36f81cb68386ec, so this doc should be updated to not use a reference.

```
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: HttpRequest
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: RouteResult
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.find_for_request:1: WARNING: py:class reference target not found: HttpRequest
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.update_aliases:1: WARNING: py:class reference target not found: optional
```

These are from type hints I think, not sure how to solve this yet.

```
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.update_aliases:1: WARNING: py:class reference target not found: PageRevision
```

`PageRevision` was renamed to `Revision`, we might be able to fix this by updating the docstring for `wagtail.models.Page.update_aliases`.

```
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.FieldBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.ChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.MultipleChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.PageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/documents/blocks.py:docstring of wagtail.documents.blocks.DocumentChooserBlock:1: WARNING: py:class reference target not found: wagtail.documents.views.chooser.BaseDocumentChooserBlock
wagtail/wagtail/images/blocks.py:docstring of wagtail.images.blocks.ImageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/snippets/blocks.py:docstring of wagtail.snippets.blocks.SnippetChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/blocks/static_block.py:docstring of wagtail.blocks.static_block.StaticBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/struct_block.py:docstring of wagtail.blocks.struct_block.StructBlock:1: WARNING: py:class reference target not found: wagtail.blocks.struct_block.BaseStructBlock
wagtail/wagtail/blocks/list_block.py:docstring of wagtail.blocks.list_block.ListBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/stream_block.py:docstring of wagtail.blocks.stream_block.StreamBlock:1: WARNING: py:class reference target not found:
```

Looks like these are referring to the undocumented base classes, so the references can be removed.

**Comment by HimanshuGarg47** [comment]
```
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_rendition:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.get_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.find_existing_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.create_renditions:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: wagtail.images.models.Filter
```
 :point_up_2: I am not able to resolve this errors. Can you help me how to fix this, I tried a lot changing noindex to noindexentry
try lot of solutions like:

- Give noindex/noindexentry  vice versa to duplicate objects
- Try to add Filter to conf.py as autodoc_type_aliases , 
but nothing works


```
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: HttpRequest
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: RouteResult
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.find_for_request:1: WARNING: py:class reference target not found: HttpRequest
```
:point_up_2: These are unknown to me. Tried various ways, but nothing works to me. I think I am not able to find root cause in detail.


```
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.FieldBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.ChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.MultipleChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.PageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/documents/blocks.py:docstring of wagtail.documents.blocks.DocumentChooserBlock:1: WARNING: py:class reference target not found: wagtail.documents.views.chooser.BaseDocumentChooserBlock
wagtail/wagtail/images/blocks.py:docstring of wagtail.images.blocks.ImageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/snippets/blocks.py:docstring of wagtail.snippets.blocks.SnippetChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/blocks/static_block.py:docstring of wagtail.blocks.static_block.StaticBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/struct_block.py:docstring of wagtail.blocks.struct_block.StructBlock:1: WARNING: py:class reference target not found: wagtail.blocks.struct_block.BaseStructBlock
wagtail/wagtail/blocks/list_block.py:docstring of wagtail.blocks.list_block.ListBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/stream_block.py:docstring of wagtail.blocks.stream_block.StreamBlock:1: WARNING: py:class reference target not found: wagtail.blocks.stream_block.BaseStreamBlock
```
:point_up_2: For above errors, I have doubt, Should I remove references to non-documented parent class/base class by remove :show-inheritence flag or any other way.

```
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: django.core.files.base.File
wagtail/wagtail/images/models.py:docstring of wagtail.images.models.AbstractImage.generate_rendition_file:1: WARNING: py:class reference target not found: django.core.files.base.File
```
Above errors, I am able to solve by adding
```
autodoc_type_aliases = {
    'File' : 'django.core.files.File',
}
``` :point_up_2:to conf.py. I also tried various ways to solve this, but this works. Did I correct?


```
wagtail/docs/releases/1.2.rst:118: WARNING: py:meth reference target not found: wagtail.models.Page.copy
```
 :point_up_2: 
  Above error, I don't know why it is not referring to method, I think it is correct?

Most above errors, I think they are because of typehinting and fast forward references. For example,
this code
https://github.com/wagtail/wagtail/blob/main/wagtail/images/models.py#L637-L639
if we remove "Filter" and return type hint, it works fine. Problem is with autodoc is unable to understand where is custom type belongs to 
and I tried googling for this issue and may be our issue is also related to sphinx autodoc extension issue. I might be wrong.

File: wagtail/models/__init__.py
Comment (HimanshuGarg47) [inline wagtail/models/__init__.py:1]: In 1295 line ``def route_for_request(request: "HttpRequest", path: str) -> RouteResult | None: `` change to 
``def route_for_request(request, path: str) -> "RouteResult" | None:``
and in 1322 line 
``    def find_for_request(request: "HttpRequest", path: str) -> "Page" | None:`` change to ``def find_for_request(request, path: str) -> "Page" | None:``

we can remove warning by this. This is solution in my mind and it is working. If it should not be that way, we need to figure out in other way.

**Comment by HimanshuGarg47** [comment]
I am able to solve 
``
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.FieldBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.ChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.MultipleChoiceBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.BaseChoiceBlock
wagtail/wagtail/blocks/field_block.py:docstring of wagtail.blocks.field_block.PageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/documents/blocks.py:docstring of wagtail.documents.blocks.DocumentChooserBlock:1: WARNING: py:class reference target not found: wagtail.documents.views.chooser.BaseDocumentChooserBlock
wagtail/wagtail/images/blocks.py:docstring of wagtail.images.blocks.ImageChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/snippets/blocks.py:docstring of wagtail.snippets.blocks.SnippetChooserBlock:1: WARNING: py:class reference target not found: wagtail.blocks.field_block.ChooserBlock
wagtail/wagtail/blocks/static_block.py:docstring of wagtail.blocks.static_block.StaticBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/struct_block.py:docstring of wagtail.blocks.struct_block.StructBlock:1: WARNING: py:class reference target not found: wagtail.blocks.struct_block.BaseStructBlock
wagtail/wagtail/blocks/list_block.py:docstring of wagtail.blocks.list_block.ListBlock:1: WARNING: py:class reference target not found: wagtail.blocks.base.Block
wagtail/wagtail/blocks/stream_block.py:docstring of wagtail.blocks.stream_block.StreamBlock:1: WARNING: py:class reference target not found: wagtail.blocks.stream_block.BaseStreamBlock
``
by removing :show-inheritence: flag in wagtail/docs/reference/streamfield/blocks.md to classes responsible for errors. Should I do this?

File: wagtail/models/__init__.py
Comment (HimanshuGarg47) [reply wagtail/models/__init__.py:1]: For these warnings 
``
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: HttpRequest
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.route_for_request:1: WARNING: py:class reference target not found: RouteResult
wagtail/wagtail/models/__init__.py:docstring of wagtail.models.Page.find_for_request:1: WARNING: py:class reference target not found: HttpRequest
``

**Comment by HimanshuGarg47** [comment]
ok @zerolab, Thanks for informing.

**Comment by zerolab** [comment]
Will close this for now, but rest assured you will be credited when #12044 is merged.

Thank you so much for your efforts!
